src/hooks/useArticlePreviewTracking.js

```diff
@@ -0,0 +1,42 @@
+import { useLayoutEffect } from 'preact/hooks'
+import { sendEvent, isInstrumentationEnabled } from 'utils'
+
+const SCHEMA_NAME = 'VirtualPageView'
+const SCHEMA_REV = 19883675
+
+export const useArticlePreviewTracking = (
+  page, source, language
+) => {
+  if (!isInstrumentationEnabled()) {
+    return
+  }
+
+  const logVirtualPageView = () => {
+    const event = {
+      /* eslint-disable camelcase */
+      access_method: 'mobile app',
+      page_title: page.titles.canonical,
+      page_namespace: page.namespace,
+      page_id: page.id,
+      source_title: source.articleTitle,
+      source_namespace: source.namespace,
+      source_page_id: source.id,
+      source_url: location.href
+      /* eslint-enable camelcase */
+    }
+
+    sendEvent(SCHEMA_NAME, SCHEMA_REV, language, event)
+  }
+
+  useLayoutEffect(() => {
+    if (!page) {
+      return
+    }
```

Comment:
so this is still needed here, the summary is a state and start with null until the api request is done

Reply:
Why `useLayoutEffect` specifically instead of `useEffect`?

Layout effect is blocking and should only be used to prevent flickering on visual effects.